### PR TITLE
add focus search bar shortcut

### PR DESCRIPTION
**Pull request by andyhuang91**
@zdhickman @gchomatas @jonathanwgoodwin

### CODE REVIEW

File: BlazarUI/app/scripts/components/shared/SearchFilter.jsx [hunk 1]
```diff
@@ -19,7 +21,7 @@ class SearchFilter extends Component {
       this.props.onChange(this.refs.searchFilterInput.value);
     }, 250);
   }
-    
+
   componentDidMount() {
     window.addEventListener('keyup', this.handleKeyup);
```
Comment (andyhuang91) [inline BlazarUI/app/scripts/components/shared/SearchFilter.jsx]: it looks like this functionality used to be in the UI but was incompletely removed:
https://github.com/HubSpot/Blazar/commit/793d52746bf8e107b6385586007ee14dde840b93

File: BlazarUI/app/scripts/components/shared/SearchFilter.jsx [hunk 2]
```diff
@@ -40,14 +42,23 @@ class SearchFilter extends Component {
...
+  handleChange() {
     this.setState({
       searchValue: this.refs.searchFilterInput.value
     });
     this.handleSearchDebounced();
   }
 
-  render() {    
+  handleKeyup(e) {
+    const notFocusedOnInputElement = e.target === document.body || e.target.tagName === 'A';
```
Comment (andyhuang91) [inline BlazarUI/app/scripts/components/shared/SearchFilter.jsx]: it's a little tricky figuring out when this shortcut should be enabled. the tagName part is needed to make sure that the shortcut still works after clicking on a sidebar link.

File: BlazarUI/app/scripts/components/shared/SearchFilter.jsx [hunk 3]
```diff
@@ -40,14 +42,23 @@ class SearchFilter extends Component {
...
+    const modifierKey = e.metaKey || e.shiftKey || event.ctrlKey;
+    const shortcutPressed = FOCUS_SEARCH_BAR_SHORTCUTS.includes(e.which);
```
Comment (andyhuang91) [inline BlazarUI/app/scripts/components/shared/SearchFilter.jsx]: Is Array.prototype.includes allowed? It's not supported in IE and babel has a polyfill for it
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Browser_compatibility

I can switch it to http://underscorejs.org/#contains if that's safer.

Comment (zdhickman) [inline BlazarUI/app/scripts/components/shared/SearchFilter.jsx]: I've been using underscore functions for the sake of consistency, but I don't think it matters much either way (hopefully none of our devs are using IE 🙊)

Maybe just use `contains` to match usage in the rest of the app  -we can swap to `includes` in the future if we want to

Comment (jonathanwgoodwin) [inline BlazarUI/app/scripts/components/shared/SearchFilter.jsx]: What Zack said + some external user may at some point try to use IE w/ blazar.

**Comment by zdhickman** [comment]
Beyond the underscore nit, LGTM 👍 thanks!

**Comment by jonathanwgoodwin** [comment]
🎉 🐬 🌊

**Pull request merged**